### ISSUE TITLE
Segfault

### ISSUE THREAD

**Issue by KenanSulayman** [centaurean/sharc#20]
Ho,

just testing this for our log rotation and it segfaulted upon compressing a gunzip tarball with the lion algorithm.

`GNU/Linux (3.16.0-29) (Little endian system, 64 bits) using GCC 4.9.1`

You can get the sample file (gunzip tarball of a git io.js clone) here: http://r2.geoca.st:65001/io.tgz

```
$ gdb --args sharc -c3 io.tgz 
...
(gdb) run
Starting program: /usr/local/bin/sharc -c3 io.tgz
File io.tgz.sharc already exists. Do you want to overwrite it (y/N) ? y

Program received signal SIGSEGV, Segmentation fault.
0x00000000004040a7 in main ()
(gdb) backtrace
#0  0x00000000004040a7 in main ()
```

Would love some insight!

Cheers

**Comment by gpnuma** [comment]
Hello, thanks for your input.

> upon compressing a gunzip tarball

Just wondering, why do you try to compress an already compressed file ? Is it just for testing ?
Lion is brand new so there are probably some glitches yet to be fixed, I'll file this as a bug and look it up - thanks for the test file BTW.

For production type compression you're better off with chameleon or cheetah at this stage.

**Comment by KenanSulayman** [comment]
Yup, just for testing. It compressed itself (the binary of sharc) to ~7% and I was so amazed I wanted to try it with an archive :-)

**Comment by gpnuma** [comment]
I see :)

**Comment by gpnuma** [comment]
Fixed in 78128cbce84bbeef1c8f1f879f00bc4a6cb82527
